deeppavlov/models/doc_retrieval/logit_ranker.py

```diff
@@ -33,50 +32,76 @@ class LogitRanker(Component):
+                ]:
 
-    def __call__(self, contexts_batch: List[List[str]], questions_batch: List[List[str]]) -> \
-            Tuple[List[str], List[float]]:
         """
         Sort obtained results from squad reader by logits and get the answer with a maximum logit.
 
         Args:
             contexts_batch: a batch of contexts which should be treated as a single batch in the outer JSON config
             questions_batch: a batch of questions which should be treated as a single batch in the outer JSON config
-
+            doc_ids_batch (optional): names of the documents from which the contexts_batch was derived
         Returns:
-            a batch of best answers and their scores
-
+             a batch of best answers, their scores, places in contexts
+             and doc_ids for this answers if doc_ids_batch were passed
         """
-        # TODO output result for top_n
-        warnings.warn(f'{self.__class__.__name__}.__call__() API will be changed in the future release.'
-                      ' Instead of returning Tuple(List[str], List[float] will return'
-                      ' Tuple(List[List[str]], List[List[float]]).', FutureWarning)
+        if doc_ids_batch is None:
+            logger.warning("you didn't pass tfidf_doc_ids as input in logit_ranker config so "
+                           "batch_best_answers_doc_ids can't be compute")
 
         batch_best_answers = []
-        batch_best_answers_scores = []
-        for contexts, questions in zip(contexts_batch, questions_batch):
+        batch_best_answers_score = []
+        batch_best_answers_place = []
+        batch_best_answers_doc_ids = []
+        for quest_ind, [contexts, questions] in enumerate(zip(contexts_batch, questions_batch)):
             results = []
             for i in range(0, len(contexts), self.batch_size):
                 c_batch = contexts[i: i + self.batch_size]
                 q_batch = questions[i: i + self.batch_size]
                 batch_predict = zip(*self.squad_model(c_batch, q_batch))
                 results += batch_predict
             if self.sort_noans:
-                results = sorted(results, key=lambda x: (x[0] != '', x[2]), reverse=True)
+                results_sort = sorted(results, key=lambda x: (x[0] != '', x[2]), reverse=True)
             else:
-                results = sorted(results, key=itemgetter(2), reverse=True)
-            batch_best_answers.append(results[0][0])
-            batch_best_answers_scores.append(results[0][2])
-        return batch_best_answers, batch_best_answers_scores
+                results_sort = sorted(results, key=itemgetter(2), reverse=True)
+            batch_best_answers.append([x[0] for x in results_sort[:self.top_n] if x[0]])
+            batch_best_answers_place.append([x[1] for x in results_sort[:self.top_n] if x[0]])
+            batch_best_answers_score.append([x[2] for x in results_sort[:self.top_n] if x[0]])
```

Comment:
remove filtering of empty answers
```suggestion
            batch_best_answers.append([x[0] for x in results_sort[:self.top_n]])
            batch_best_answers_place.append([x[1] for x in results_sort[:self.top_n]])
            batch_best_answers_score.append([x[2] for x in results_sort[:self.top_n]])
```